backend/tursu_admin/views.py

```diff
@@ -64,3 +65,30 @@ def verify_product(request):
     product.is_verified = True
     product.save()
     return HttpResponse("Successfully verified product")
+
+@authentication_classes([SessionAuthentication, BasicAuthentication])
+@permission_classes((IsAuthenticated,))
+@api_view(['POST'])
+def ban_user(request):
+    """Bans given user."""
+    user = get_vendor_from_request(request)
+    if user == None:
+        user = get_customer_from_request(request)
+        if user == None:
+            return HttpResponse("There is no such user.", status=400)
+    user.user.is_banned = True
+    user.save()
```

Comment:
user.user has a field changed here but user is saved. user.user should be saved instead. Also the variable naming can be changed for cleaner code.

Reply:
Necessary corrections are made.